### ISSUE TITLE
TensorBoard Has Some Odd Behavior?

### ISSUE THREAD

**Issue by ljarasunas2021** [Unity-Technologies/ml-agents#3743]
Hello,

I seem to be getting some really odd behavior with machine learning (gail + bc + curiosity + rl) and especially with tensor board.  First off, here are the hyper-parameters of my gail_config.yaml:

<img width="380" alt="Screen Shot 2020-04-04 at 7 21 26 PM" src="https://user-images.githubusercontent.com/43383815/78465363-8cae3880-76a9-11ea-9304-982bbc4a477a.png">

and I run this to train my the ml agents:

`mlagents-learn config/gail_config.yaml --env=/Users/lukas/Desktop/test356.app --train`

Please let me know if there is anything inherently wrong with either of these that may cause the following bugs/errors.  Ok, so here's the interesting part.  Firstly, my agent does not seem to be improving (the mean reward fluctuates a lot).  Just as an example, my previous trial led to this behavior, listed in chronological order:

`(step, mean reward): 
(5000, 4.911)
(8000, 3.015)
(9000, 0.198)
(11000, 0.060)
(13000, 2.634)
(14000, 0.998)
(15000, -0.681)
(16000, 1.295)
(17000, 1.066)
(18000, -0.284)
(20000, 0.816)
(21000, -0.543)
(23000, 3.050)
(26000, 2.441)
(29000, 4.729)
`

Looking at this data, at the end it seems to start increasing a lot, but I did just end it as it did not exactly look like it was ever going to coverge to a large mean reward.  Also, note that the somewhat complex behavior of my game may cause that variation (I don't actually know if it does - I'm a little new to machine learning in Unity).  Should I try running it out longer? Simplifying the gail_config to not use (gail, bc, rl, and curiosity)? Any other suggestions?  Oh and I need to mention that I'm just trying to get a basic behavior currently, nothing fancy, and then I will scale up the vector observations and other parameters once I can get some basic behavior working.  

So now comes the really weird part and the part that this bug is mainly centered around (although it can be due to any of the information I have shared previously).  So after running the tensor board, my graphs look like this with a smoothing of 0:

<img width="620" alt="Screen Shot 2020-04-04 at 7 15 26 PM" src="https://user-images.githubusercontent.com/43383815/78465532-b1a3ab00-76ab-11ea-87a5-55313f73b462.png">

<img width="618" alt="Screen Shot 2020-04-04 at 7 15 41 PM" src="https://user-images.githubusercontent.com/43383815/78465533-b6685f00-76ab-11ea-956c-9e73ee871091.png">

So even though I got to 33000 steps, the first two graphs only show 15000 steps in total, which is odd.  Secondly, they do not behave as functions, or for some inputs there are multiple outputs, which doesn't really make sense to me since there can't be two different values (like cumulative reward) at a since point in time.  This is even more crazy with the loss graphs.  Also, the values on the graphs do not seem to be accurate when compared to the values shown above.  

Now, even weirder, when I smooth the graphs (to point 5 in this instance), I get totally different graphs: 

<img width="616" alt="Screen Shot 2020-04-04 at 7 15 58 PM" src="https://user-images.githubusercontent.com/43383815/78465584-5faf5500-76ac-11ea-83f9-c9f7e22150f9.png">

<img width="615" alt="Screen Shot 2020-04-04 at 7 16 08 PM" src="https://user-images.githubusercontent.com/43383815/78465585-6342dc00-76ac-11ea-9aeb-985f465962aa.png">
 
Why would these smooth graphs be so different? In fact the cumulative reward is now shows a decreasing trend unlike the increasing trend when the smoothing was 0.  I would really appreciate it if someone could make tell me how to approach this issue (the tensorboard graphs, and more importantly the machine learning in general).  Thank you so much!!!

EDIT: My current MLAgents version is 12.0 if you have made significant changes to Tensorboard in the latest releases.  I would use version 15.1 or 15.0, but both of those version return errors upon importing them (not errors in the code I wrote but errors in the actual docs of mlagents), so instead of trying to fix those bugs, I just went back to my preliminary version of ml agents.

**Comment by awjuliani** [comment]
Hi @ljarasunas2021 

The issue with the tensorboard graphs being rendered strangely is due to you likely not using a unique `run-id` for each training session. If you don't use a novel `run-id` then tensorboard will combine the data collected from every run into a single graph. Try calling `mlagents-learn` with a novel `run-id=something`. Let me know if that resolves your issue.

**Comment by ljarasunas2021** [comment]
ah thank you will try that

**Comment by dongruoping** [comment]
Closing the issue due to inactivity and no obvious bug. 
If your issue wasn't resolved, please feel free to re-open it.
